### PR TITLE
Resolve Hashable related warnings

### PR DESCRIPTION
**Pull request by hartbit**
Took the opportunity to resolve a few FIXMEs that still did not use Swift 4.2's automatic Hashable synthesis.

### CODE REVIEW

**Comment by hartbit** [comment]
@lorentey In case you want to check this out.

File: Sources/Utility/ArgumentParser.swift [hunk 1]
```diff
@@ -290,8 +290,8 @@ protocol ArgumentProtocol: Hashable {
 extension ArgumentProtocol {
     // MARK: - Conformance for Hashable
 
-    public var hashValue: Int {
-        return name.hashValue
+    public func hash(into hasher: inout Hasher) {
```
[the remark this answers is not in the document]
Comment (hartbit) [reply Sources/Utility/ArgumentParser.swift]: `hash(into:)` was implemented in 4.2: https://github.com/apple/swift-evolution/blob/master/proposals/0206-hashable-enhancements.md

Btw, do we still support 4.0?

[the remark this answers is not in the document]
Comment (aciidb0mb3r) [reply Sources/Utility/ArgumentParser.swift]: Basic/Utility needs to support the latest GM so we can drop 4.0 and just adopt 4.2

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Pull request merged**